docs/quick-start.md

```diff
@@ -1,28 +1,121 @@
+
+Numalogic can be used as an independent library, it provides various ML models and tools. Here, we are using a `SparseAEPipeline`, you can refer to [training section](autoencoders.md) for other available options. 
+
+In this example, we have numbers ranging from 1-10 in the train data set. In the test data set, we have data points that go out of this range, which the algorithm should be able to detect as anomalies.
+
+```python
+from numalogic.models.autoencoder import SparseAEPipeline
+from numalogic.models.autoencoder.variants import Conv1dAE
+from numalogic.scores import tanh_norm
+
+X_train = [[1], [3], [5], [2], [5], [1], [4], [5], [1], [4], [5], [8], [9], [1], [2], [4], [5], [1], [3]]
+X_test = [[-20], [3], [5], [40], [5], [10], [4], [5], [100]]
 
-Run the following command to install the dependencies and build locally.
+model = SparseAEPipeline(
+    model=Conv1dAE(in_channels=1, enc_channels=4), seq_len=8, num_epochs=30
+)
+# fit method trains the model on train data set
+model.fit(X_train)
+
+# predict method returns the reconstruction error
+recon = model.predict(X_test)
+
+# score method returns the anomaly score computed on test data set
+anomaly_score = model.score(X_test)
+
+# normalizing scores to range between 0-10
+anomaly_score_norm = tanh_norm(anomaly_score)
+print("Anomaly Scores:", anomaly_score_norm)
+```
+
+Below is the sample output, which has logs and anomaly scores printed. You can notice the anomaly score for points -20, 40 and 100 in `X_test` is high.
 ```shell
-#install poetry
-curl -sSL https://install.python-poetry.org | python3 -
+2022-10-19 17:37:08,241 - INFO - Current device: cpu
+2022-10-19 17:37:08,241 - INFO - Current device: cpu
+2022-10-19 17:37:08,245 - INFO - Training sparse autoencoder model with beta: 0.001, and rho: 0.05
+2022-10-19 17:37:08,245 - INFO - Using kl_div regularized loss
+2022-10-19 17:37:08,274 - INFO - epoch : 5, penalty: 0.00017940076941158623 loss_mean : 1.6290421
+2022-10-19 17:37:08,282 - INFO - epoch : 10, penalty: 0.0001806353684514761 loss_mean : 1.5867264
+2022-10-19 17:37:08,290 - INFO - epoch : 15, penalty: 0.00018205904052592814 loss_mean : 1.5468330
+2022-10-19 17:37:08,297 - INFO - epoch : 20, penalty: 0.0001839457399910316 loss_mean : 1.5095336
+2022-10-19 17:37:08,306 - INFO - epoch : 25, penalty: 0.00018673710292205215 loss_mean : 1.4750170
+2022-10-19 17:37:08,314 - INFO - epoch : 30, penalty: 0.0001903115917230025 loss_mean : 1.4423900
+Anomaly Scores: [[3.43350191]
+ [1.0774759 ]
+ [0.81873476]
+ [5.34637364]
+ [1.38783935]
+ [0.3643268 ]
+ [1.14278396]
+ [2.78489574]
+ [7.41677152]]
+```
+
+Replace `X_train` and `X_test` with your own data, and see the anomaly scores generated.
 
-#activate virtual env
-poetry shell
+For more detailed experimentation, refer to [quick-start-example](https://github.com/numaproj/numalogic/blob/main/examples/quick-start.ipynb)
 
-#install dependencies
-poetry install
+## Numalogic as streaming ML using Numaflow
```

Comment:
this section is not self-explanatory; we need to add a few points we discussed during today's meeting. Can we add those?

Reply:
Does this sound okay? Should I include a DAG picture for pipeline?

Numalogic can also be paired with our DAG based streaming platform [Numaflow](https://numaflow.numaproj.io/), to build streaming ML pipelines where Numalogic can be used in UDF.